### PR TITLE
Added Kademlia mode check before inserting peer to routing table

### PR DESCRIPTION
**Pull request by sh3ll3x3c**
- Identify protocols  `agent_version` is used for propagating LCs Kademlia mode
- On identify received event, only peers in `server` mode are inserted to the routing table
- Switched the default Kademlia mode to `client`
- `agent_version` is now in the `{base_string}/{client_type}/{kademlia_mode}` format

### CODE REVIEW

File: src/network/p2p/event_loop.rs [hunk 1]
```diff
@@ -304,47 +304,58 @@ impl EventLoop {
...
+							listen_addrs,
+							agent_version,
+							protocol_version,
+							..
+						},
+				} => {
+					trace!(
+						"Identity Received from: {peer_id:?} on listen address: {listen_addrs:?}"
+					);
+					let incoming_peer_agent_version = match AgentVersion::from_str(&agent_version) {
```
Comment (momosh-ethernal) [inline src/network/p2p/event_loop.rs:323]: use let else here?

File: src/network/p2p/event_loop.rs [hunk 2]
```diff
@@ -304,47 +304,58 @@ impl EventLoop {
...
+						Ok(agent) => agent,
+						Err(e) => {
+							info!("Error parsing incoming agent version: {e}");
+							return;
+						},
+					};
+					if protocol_version != self.identity_data.protocol_version {
```
Comment (momosh-ethernal) [inline src/network/p2p/event_loop.rs:330]: reverse condition? math if without negation

**Review by momosh-ethernal — APPROVED** [review]

File: src/network/p2p/event_loop.rs (hunk 1 above)
Comment (sh3ll3x3c) [reply src/network/p2p/event_loop.rs:323]: We need both values in this case, so a match is probably a better choice...

File: src/network/p2p/event_loop.rs (hunk 2 above)
Comment (sh3ll3x3c) [reply src/network/p2p/event_loop.rs]: Done.

**Review by luka-ethernal — APPROVED** [review]

**Pull request merged**